apps/smart-contracts/core/test/AllowedCallers.test.ts

```diff
@@ -0,0 +1,37 @@
+import chai, { expect } from 'chai'
+import { ethers } from 'hardhat'
+import { SignerWithAddress } from '@nomiclabs/hardhat-ethers/dist/src/signer-with-address'
+import { smock } from '@defi-wonderland/smock'
+import { allowedCallersFixture } from './fixtures/AllowedCallersFixture'
+import { AllowedCallers } from '../typechain'
+
+chai.use(smock.matchers)
+
+describe('=> AllowedCallers', () => {
+  let restrictedCallers: AllowedCallers
+  let deployer: SignerWithAddress
+  let user: SignerWithAddress
+
+  beforeEach(async () => {
+    ;[deployer, user] = await ethers.getSigners()
+    restrictedCallers = await allowedCallersFixture()
+  })
+
+  describe('initial state', () => {
+    it('has no allowed accounts', async () => {})
+  })
+
+  describe('# setAllowedCallers', () => {
+    it('should revert when callers.length != allowed.length', async () => {})
```

Comment:
```suggestion
    it('reverts if callers.length != allowed.length', async () => {})
```
Remember, we do not use should wording for testcases, and also not "when" but "if".

Please revise the other testcase names as well.